### ISSUE TITLE
Loopback Devicetree Troubles

### ISSUE THREAD

**Issue by dominicharkness** [jeremytrimble/ezdma#6]
I'm not sure if this is the right place to ask this, but I wasn't able to find your email anywhere. 

Anyway, we'd like to use this driver to stream data from the PL on the Zedboard into the linux userspace. We are initially trying to get your loopback example running on the Zedboard, however we're having trouble getting the devicetree working. Currently we get the following error:

```
ezdma: allocated chrdev region: Major: 245, Minor: 0-8
probing ezdma
ezdma: couldn't find dma channel: loop_tx, deferring...
ezdma: loop_tx (TX) available
ezdma: couldn't find dma channel: loop_rx, deferring...
ezdma: loop_rx (RX) available
platform ezdma0: Driver ezdma requests probe deferral
```

Our dma device in the device tree looks like:
```
/ {
	amba_pl: amba_pl {
		#address-cells = <1>;
		#size-cells = <1>;
		compatible = "simple-bus";
		ranges ;
		axi_dma_1: axidma@40400000 {
			#dma-cells = <1>;
			compatible = "xlnx,axi-dma";
			interrupt-parent = <&intc>;
			interrupts = <0 29 4 0 30 4>;
			reg = <0x40400000 0x10000>;
			xlnx,include-sg ;
			loop_tx: dma-channel@40400000 {
				compatible = "xlnx,axi-dma-mm2s-channel";
				interrupts = <0 29 4>;
				xlnx,datawidth = <0x20>;
				xlnx,device-id = <0x1>;
			};
			loop_rx: dma-channel@40400030 {
				compatible = "xlnx,axi-dma-s2mm-channel";
				interrupts = <0 30 4>;
				xlnx,datawidth = <0x20>;
				xlnx,device-id = <0x1>;
			};
		};
	};
	ezdma0 {
        compatible = "ezdma";

        dmas = <&axi_dma_1 0 &axi_dma_1 1>;
        dma-names = "loop_tx", "loop_rx";   // used when obtaining reference to above DMA core using dma_request_slave_channel()
        ezdma,dirs = <2 1>;                 // direction of DMA channel: 1 = RX (dev->cpu), 2 = TX (cpu->dev)
    };
};
```
I was hoping you may be able to help figure out why the driver can't find the device as we defined in our devicetree. If this is the wrong forum for this issue please let me know how I should contact you. Thanks!!

**Comment by dominicharkness** [comment]
Just wanted to bring some resolution to this. We found that the issues was a combination of the wrong linux kernel version and errors in the devicetree. We found your linux-xlnx repo and used that version of the kernel (4.x) along with changing `compatible = "xlnx,axi-dma";` to `compatible = "xlnx,axi-dma-1.00.a";`

**Comment by raffaelesury** [comment]
hi dominic, did you use petalinux? 

Did you have success in running the example and your application?

**Comment by dominicharkness** [comment]
We did not use petalinux, however we were able to successfully run the example.

**Comment by DarrenChengdu** [comment]
Hi Dominic,

I encounter the same output just like "ezdma: allocated chrdev region: Major: 244 (not 245 as yours), Minor: 0-8", by using Petalinux 2016.3. I plan to generate ezdma.ko module not using petalinux. I would really appreciate it if you post the xlnx-linux and u-boot versions in your work.

**Comment by noelpedro** [comment]
On zedboard Got loopRX and loopTX available using petalinux 2018.3 I just added custom ezdma kernel module and added ezdma nodes to system-user.dtsi
